[PkgConfigDeps][MesonToolchain] Added build context folder mechanism (#3640)
conan-io/docs: reference/tools/gnu/pkgconfigdeps.rst

build_context_suffix
^^^^^^^^^^^^^^^^^^^^

> *DEPRECATED: use build_context_folder attribute instead*

Reviewer: Is this true for ``autotools`` projects too? Or only for ``Meson``? I think only Meson has the capability of specifying a folder for the build-requires?

cross-building because the tool (that will run in the building machine) belongs to a
different binary package than the library, that will "run" in the host machine.

You can use the ``build_context_folder`` attribute to specify a folder to save the `*.pc` files created by all those
build requirements listed in the ``build_context_activated`` one:

.. code-block:: python

    tool_requires = ["my_tool/0.0.1"]
    requires = ["my_tool/0.0.1"]
    def generate(self):
        pc = PkgConfigDeps(self)
        # generate the *.pc file for the tool require
        pc.build_context_activated = ["my_tool"]
        # save all the *.pc files coming from the "my_tool" build context and its requirements
        pc.build_context_folder = "build"  # [generators_folder]/build/
        pc.generate()

build_context_suffix
^^^^^^^^^^^^^^^^^^^^

> *DEPRECATED: use build_context_folder attribute instead*

Reviewer: It is still not very clear that the deprecation of this and the recommendation to use ``build_context_suffix`` is the way to go also for autotools.  If it is not, maybe we should recommend this for Meson only?

Maybe we need to understand the potential autotools usages of the ``build_context_suffix`` and ``build_context_folder``

        pc.generate()

build_context_folder
^^^^^^^^^^^^^^^^^^^^

*New since Conan 2.2.0*

Same concept as the quoted ``build_context_suffix`` attribute above, but this is meant to specify a folder to save

> the `*.pc` files created by all those build requirements listed in the ``build_context_activated`` one:

Reviewer 2: I think it would be better to not reference the deprecated attribute to document this one, just explaining this as if the other one did not exist as one day we will remove that one.

Author: Reordered

        pc.generate()

.. important::

>     You cannot use this attribute along together with the ``build_context_folder`` one.

Reviewer 2: ```suggestion
    This attribute should not be used simultaneously with the ``build_context_folder`` attribute.
```
